### PR TITLE
Creates a header with a logo

### LINKED ISSUE #28: 14. Create logo for header
## Summary

Create a simple shopping cart logo including the app name and image/icon. 

## Acceptance criteria
- [x] Ensure file format is compatible (eg. SVG)
- [x] Ensure background is transparent
- [x] Ensure file size is large enough so it's not pixelated when enlarged on screen
 
## Notes
- Canva is a good tool for designing logos

### LINKED ISSUE #29: 15. Create header with logo and navigation items
## Summary

Create a header at the top of the page with the logo (created in Issue 14) on the left side and navigation items on the right side. The navigation items should include: Home, List, and List Item.  

## Acceptance criteria
- [x] Header has logo on left side and nav items are on right side.

### PR DESCRIPTION
**Pull request by avinashi10**
## Description

This code updates the title and navigation into a single header at the top of the page. It adds a logo, a background, and updates the navigation tabs to match the color scheme.

## Related Issue

Closes #28 & Closes #29 

## Acceptance Criteria

- [x] Ensure file format is compatible (eg. SVG)
- [x] Ensure background is transparent
- [x] Ensure file size is large enough so it's not pixelated when enlarged on screen
- [x] Header has logo on left side and nav items are on right side. 

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|    | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |
| ✓   | :sparkling_heart: Styling     |

### Before

Screenshot 2023-10-02 at 6 13 00 PM

### After

Screenshot 2023-10-02 at 6 12 08 PM

## Testing Steps / QA Criteria

1. Switch to the `logo-header` branch on your local computer
2. Run `npm i` to install the newest dependencies
4. In your web browser view the page and ensure the custom header pictured above is loading.
5. Click the navigation tabs to make sure the work.
6. Navigate to the 'Add Item' page. Click the logo and verify it navigates you back to the home view (which should be the list page).

### CODE REVIEW

**Review by MagdaSlifierz — APPROVED** [review]
The logo looks really nice. Thanks

**Comment by hajschroeder** [comment]
Looks great! 

Just a little input:
- The logo is spectacular, but the text is a little hard to read. Would it make more sense to have the content just say "PC" or a different abbreviation for the insignia,  and have the name of the app in larger text somewhere else in the app, such as the homepage? Side note: very good call on associating a NavLink with the shopping cart logo. 
- Also in regards to text sizing, the words within the nav Tab links are very small. It looks great on mobile, but is very sparse and tiny on desktop. Even with my glasses on, I needed to lean closer to my monitor to see how the header was laid out and what the text within the Tab links said. Since we are focused on accessibility, I am not sure the best course of action to improve this, but I wanted to bring that up.

Nicely done!

File: src/images/PredictaCartLogoAltColor.png
Comment (hajschroeder) [inline src/images/PredictaCartLogoAltColor.png:1]: The clocks as wheels in the logo is a great touch!

**Review by codecaviette — COMMENTED** [review]
All of the ACs have been met and the logo design looks clean! 

A couple of notes: 
1. The text of the logo is a bit small compared to the size of the logo, making it difficult to read. We could consider making the text larger, overlaying it on the cart shape, and creating enough contrast so the text is easily readable. Or, we could make the text larger and set it beside the cart shape. Just some ideas!
2. When clicking on the logo from the Add Item page, the page does correctly navigate to the List page; however,  the "Add Item" navigation tab is still selected. Intuitively, the "List" tab should be highlighted instead.

File: src/images/PredictaCartLogoAltColor.png
Comment (avinashi10) [reply src/images/PredictaCartLogoAltColor.png:1]: thanks - glad you noticed that 🙃

**Comment by avinashi10** [comment]
@hajschroeder @codecaviette I agree on the sizing - I redesigned the logo and I increased the tab font size and spacing
@codecaviette great catch on that tab selection bug!!! I learned some new things about React Router + some built-in JS stuff in solving it. I added what I learned in the code comments of my latest commit!

File: src/views/Layout.jsx [hunk 1]
```diff
@@ -15,6 +15,23 @@ import {
 import logo from '../images/PredictaCartLogo.png';
 
 export function Layout() {
+	const location = useLocation();
```
Comment (avinashi10) [inline src/views/Layout.jsx:18]: `useLocation`: This is a hook provided by React Router, which returns the current location object that represents the current URL. You can think of it as a way of programmatically accessing the current browser URL's details.

File: src/views/Layout.jsx [hunk 2]
```diff
@@ -15,6 +15,23 @@ import {
...
+	let tabIndex;
+
+	switch (location.pathname) {
```
Comment (avinashi10) [inline src/views/Layout.jsx:21]: `location.pathname`: The location object returned by `useLocation` has various properties. One of these is `pathname`, which is a string representing the current URL's path. 

For example, if the current URL is `https://example.com/list`, then `location.pathname` would be `/list`.

The `switch` statement is a control flow statement used in JavaScript (and many other programming languages) that allows for a variable to be tested for equality against a list of values. Each possible case of the `switch` statement is represented by the `case` keyword.

File: src/views/Layout.jsx [hunk 3]
```diff
@@ -15,6 +15,23 @@ import {
...
+			tabIndex = 1;
+			break;
+		case '/add-item':
+			tabIndex = 2;
+			break;
+		default:
+			tabIndex = 0;
+	}
+
 	return (
```
Comment (avinashi10) [inline src/views/Layout.jsx:35]: - The `switch` is evaluating the value of `location.pathname`.
  
- `case '/':` checks if `location.pathname` equals `'/'`. If it does, the code within that case runs, setting `tabIndex` to `0`, and then the `break` statement ensures the switch statement stops executing.

- Similarly, `case '/list':` checks if `location.pathname` equals `'/list'`. If it does, the code within that case runs, setting `tabIndex` to `1`, and then breaks out of the switch.

- `case '/add-item':` does the same thing, checking against `'/add-item'` and potentially setting `tabIndex` to `2`.

- The `default` case is a catch-all. If none of the above cases match the value of `location.pathname`, the code inside the `default` case runs. Here, we're setting the `tabIndex` back to `0`.

The purpose of this logic is to determine which tab should be active based on the current URL path. By figuring out the active tab's index, we can then pass it to the Chakra `Tabs` component to programmatically highlight the correct tab. This ensures that when a user navigates using the logo (or any other means), the tab highlighting remains consistent with the current view.

**Review by codecaviette — APPROVED** [review]
Logo text is more legible! 

Thank you for sharing the knowledge around updating the selected tab.

**Review by thetrend — APPROVED** [review]

**Comment by lyleschemmerling** [comment]
The new list button was a little hard to find at the bottom, could be a little bigger or more obvious, with maybe a margin a the bottom

**Review by lyleschemmerling — APPROVED** [review]

**Comment by lyleschemmerling** [comment]
nevermind, just saw the other pr, heh...

**Pull request merged**